sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -500,7 +501,11 @@ object JdbcUtils extends Logging with SQLConfHelper {
 
     case TimestampType =>
       (rs: ResultSet, row: InternalRow, pos: Int) =>
-        val t = rs.getTimestamp(pos + 1)
+        val t = dialect.getDatabaseCalendar match {
+          case Some(cal) => rs.getTimestamp(pos + 1, cal)
```

Comment:
shall we always call `get/setTimestamp` with a calendar instance of the session local timezone? We can add a config to restore the old behavior, instead of adding a new API in Dialect.

Reply:
Unforntunatelly it really depends on driver implementation whether they handle difference between TZ and NTZ in their getTimestamp/setTimestamp codepath